### ISSUE TITLE
Logging inconsistencies while using Tensorboard

### ISSUE THREAD

**Issue by Fed03** [PyTorchLightning/pytorch-lightning#7705]
## 🐛 Bug

There are inconsistencies between what gets logged and what gets displayed in Tensorboard.

In the BoringModel attached below, Tensorboard displays only the dict logged during validation.
The exact same code on my machine produces a different result. Tensorboard shows the dics logged on both validation and test but forgets the train one. In addition the loss is never displayed.
With the real model I'm working to, Tensorboard perfectly displays the logs for train and validation but the test is nowhere to be seen.

## Please reproduce using the BoringModel
It's runnable. At the end it can be seen what gets logged and what Tensorboard shows.
https://colab.research.google.com/drive/1YLimzheZQtu6BrOjEoU8Lrd_Ajs0vltw?usp=sharing

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1YLimzheZQtu6BrOjEoU8Lrd_Ajs0vltw?usp=sharing).

To simulate the metric inputs, I defined a dummy_results func [here](https://colab.research.google.com/drive/1YLimzheZQtu6BrOjEoU8Lrd_Ajs0vltw#scrollTo=U7rXKuVnP0lG&line=2&uniqifier=1)

### Expected behavior

I expect that what is logged gets correctly displayed within Tensorboard

### Environment

* CUDA:
	- GPU:
		- Tesla K80
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu101
	- pytorch-lightning: 1.3.2
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Tue Apr 20 19:55:43 PDT 2021

**Comment by awaelchli** [comment]
Change 

```python
    trainer = Trainer(
        max_epochs=1, 
        progress_bar_refresh_rate=20,
    )
```

to 

```python
    trainer = Trainer(
        max_epochs=1, 
        progress_bar_refresh_rate=20,
        log_every_n_steps=1,  # or just a number that is smaller than your dataloader size / epoch size
    )
```

**Comment by Fed03** [comment]
Tried in the above linked colab, and it seems that your solution solves the train and validation logs but not the test one, that is the real problem I'm trying to resolve in my model.
 
As a side note, I think that during *_epoch_end, lightning should log regardless of log_every_n_steps value
